### PR TITLE
Blood Defines and custom blood

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
Changes most references to blood colour to use defines rather than repeat individual definitions of hex codes.
Also adds the ability for carbon mobs to have their blood changed to any reagentid using the special_blood var.
Lastly, updated the proc for xeno blood IDs to use a caste variable rather than hardcode for queen and praetorian.

# Explain why it's good for the game

# Testing Photographs and Procedure

Tested ingame.
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.
image

</details>

# Changelog

:cl:
add: Added the ability for carbon mobs to have special blood.
code: Updated the get_blood_id proc for xenos to check a variable rather than hard-code overrides for Queen and Praetorian.
code: Changed most references to blood color to use defines.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 1]
```diff
@@ -463,7 +463,13 @@
...
 
 /datum/species/proc/handle_blood_splatter(mob/living/carbon/human/human, splatter_dir)
-	var/obj/effect/temp_visual/dir_setting/bloodsplatter/bloodsplatter = new bloodsplatter_type(human.loc, splatter_dir)
+	var/color_override
+	if(human.special_blood)
+		var/datum/reagent/D = chemical_reagents_list[human.special_blood]
+		if(D)
+			color_override = D.color
+
+	var/obj/effect/temp_visual/dir_setting/bloodsplatter/bloodsplatter = new bloodsplatter_type(human.loc, splatter_dir, color_override)
```
Comment (Segrain) [inline code/modules/mob/living/carbon/human/species/species.dm:472]:
```suggestion
	var/obj/effect/temp_visual/dir_setting/bloodsplatter/bloodsplatter = new bloodsplatter_type(human.loc, splatter_dir, color_override = color_override)
```

File: code/modules/mob/living/blood.dm [hunk 2]
```diff
@@ -245,10 +245,10 @@
 
 //returns the color of the mob's blood
 /mob/living/proc/get_blood_color()
-	return "#A10808"
+	return BLOOD_COLOR_HUMAN
 
 /mob/living/carbon/xenomorph/get_blood_color()
-	return "#dffc00"
+	return BLOOD_COLOR_XENO
```
Comment (Drulikar) [inline code/modules/mob/living/blood.dm:251]: Did we not want to disambiguate royal blood from regular blood color here? Shooting a queen makes decals colored as BLOOD_COLOR_XENO

**Review by Drulikar — APPROVED** [review]
Changes appear to be working and fine, though note other comment where maybe it was unintentional to still be using a xeno blood color.

**Comment by Drulikar** [comment]
Looks like get_blood_color isn't used for the blood decals - those are just using their default color values currently. If we wanted those to change too we'd probably need to add:
`XB.color = get_blood_color()` to `/mob/living/carbon/xenomorph/add_splatter_floor(turf/T, small_drip, b_color)`
image

**Pull request merged**